node/lib/util/reset.js

```diff
@@ -60,10 +62,59 @@ function getType(type) {
         case TYPE.SOFT : return NodeGit.Reset.TYPE.SOFT;
         case TYPE.MIXED: return NodeGit.Reset.TYPE.MIXED;
         case TYPE.HARD : return NodeGit.Reset.TYPE.HARD;
+
+        // TODO: real implementation of `reset --merge`.  For now, this behaves
```

Comment:
maybe instead just throw an exception?  I've never heard of anyone using --merge.

Reply:
You don't use it directly, but it's what the rebase/merge/cherry-pick `--abort` commands use to restore your repo.

```
           --merge
               Resets the index and updates the files in the working tree that are different between <commit> and HEAD, but keeps those which are
               different between the index and working tree (i.e. which have changes which have not been added). If a file that is different between
               <commit> and the index has unstaged changes, reset is aborted.
```